userguide/tasks.rst

```diff
@@ -1676,18 +1676,36 @@ shows the configuration options for Cloud Syncs.
    +---------------------+----------------+------------------------------------------------------------------------------------------------------------+
    | Post-script         | string         | A script to execute after the Cloud Sync Task is run.                                                      |
    +---------------------+----------------+------------------------------------------------------------------------------------------------------------+
-   | Remote Encryption   | checkbox       | Encrypt files before transfer and store the encrypted files on the remote system.                          |
-   |                     |                | `rclone Crypt <https://rclone.org/crypt/>`__ is used.                                                      |
+   | Remote Encryption   | checkbox       | Use `rclone Crypt <https://rclone.org/crypt/>`__ to manage data encryption during *PUSH* or *PULL*         |
+   |                     |                | transfers:                                                                                                 |
+   |                     |                |                                                                                                            |
+   |                     |                | *PUSH:* Encrypt files before transfer and store the encrypted files on the remote system. Files are        |
+   |                     |                | encrypted using the :guilabel:`Encryption Password` and :guilabel:`Encryption Salt` values.                |
+   |                     |                |                                                                                                            |
+   |                     |                | *PULL:* Decrypt files that are being stored on the remote system before the transfer. Transferring the     |
+   |                     |                | encrypted files requires entering the same :guilabel:`Encryption Password` and :guilabel:`Encryption Salt` |
+   |                     |                | that was used to encrypt the files.                                                                        |
+   |                     |                |                                                                                                            |
+   |                     |                | Adds the :guilabel:`Filename Encryption`, :guilabel:`Encryption Password`, and :guilabel:`Encryption Salt` |
+   |                     |                | options.                                                                                                   |
```

Comment:
I think it would be fair and useful to add this link: https://rclone.org/crypt/#file-formats so users can understand what key derivation and encryption algorithms are used.